### PR TITLE
Introduce ObjectMother class for test data generation

### PR DESCRIPTION
**Pull request by kepinskik**
that generates common data for Blog and BlogPost test objects. Easy data customization is allowed via passing an optional Map parameter to factory methods.

Example usage shown on two tests: BlogPostRepositorySpec and BlogSpec

### CODE REVIEW

File: src/test/groovy/com/jvm_bloggers/ObjectMother.groovy [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+class ObjectMother {
+
+    static Blog aBlog(Map params = [:]) {
+        String id = UUID.randomUUID().toString()
+        [
+                bookmarkableId    : "bookmarkableId $id".subSequence(0,29),
+                author            : 'author',
+                rss               : "rss $id",
+                url               : "url $id",
```
Comment (tdziurko) [inline src/test/groovy/com/jvm_bloggers/ObjectMother.groovy]: At least let's make url and rss looking like valid :) e.g. `http://example.com/$id` etc.

File: src/test/groovy/com/jvm_bloggers/ObjectMother.groovy [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        ] + params as Blog
+    }
+
+    static BlogPost aBlogPost(Map params = [:]) {
+        String id = UUID.randomUUID().toString()
+        [
+                publishedDate   : LocalDateTime.now(),
+                blog            : aBlog(),
+                title           : "title $id",
+                url             : "url $id",
```
Comment (tdziurko) [inline src/test/groovy/com/jvm_bloggers/ObjectMother.groovy]: Same here

File: src/test/groovy/com/jvm_bloggers/entities/blog/BlogSpec.groovy [hunk 3]
```diff
@@ -1,25 +1,18 @@
...
-        Blog blog = Blog.builder()
-            .bookmarkableId("john-doe")
-            .author("John Doe")
-            .rss("http://john-doe.com/feed/")
-            .url("url")
-            .dateAdded(new NowProvider().now())
-            .blogType(BlogType.PERSONAL)
-            .moderationRequired(moderationRequiredValue)
-            .build()
+        Blog blog = aBlog(moderationRequired: moderationRequiredValue)
```
Comment (tdziurko) [inline src/test/groovy/com/jvm_bloggers/entities/blog/BlogSpec.groovy]: This looks awesome :)

File: src/test/groovy/com/jvm_bloggers/entities/blog_post/BlogPostRepositorySpec.groovy [hunk 4]
```diff
@@ -35,29 +35,29 @@ class BlogPostRepositorySpec extends SpringContextAwareSpecification {
...
         List<BlogPost> blogPosts = [
-            aBlogPost(1, LocalDateTime.of(2016, 1, 1, 12, 00), REJECTED, blog),
-            aBlogPost(2, LocalDateTime.of(2016, 1, 4, 12, 00), REJECTED, blog),
-            aBlogPost(3, LocalDateTime.of(2016, 1, 2, 12, 00), APPROVED, blog),
-            aBlogPost(4, LocalDateTime.of(2016, 1, 5, 12, 00), APPROVED, blog),
-            aBlogPost(5, LocalDateTime.of(2016, 1, 3, 12, 00), NOT_MODERATED, blog),
-            aBlogPost(6, LocalDateTime.of(2016, 1, 6, 12, 00), NOT_MODERATED, blog)
+            aBlogPost(publishedDate: baseTime.plusDays(0), approved: REJECTED, blog: blog),
+            aBlogPost(publishedDate: baseTime.plusDays(1), approved: REJECTED, blog: blog),
+            aBlogPost(publishedDate: baseTime.plusDays(2), approved: APPROVED, blog: blog),
```
Comment (tdziurko) [inline src/test/groovy/com/jvm_bloggers/entities/blog_post/BlogPostRepositorySpec.groovy]: For approved ones, we do not need to override the default value

File: src/test/groovy/com/jvm_bloggers/entities/blog_post/BlogPostRepositorySpec.groovy [hunk 5]
```diff
@@ -35,29 +35,29 @@ class BlogPostRepositorySpec extends SpringContextAwareSpecification {
...
         ]
 
         blogPostRepository.saveAll(blogPosts)
 
         when:
         List<BlogPost> latestPosts = blogPostRepository.findLatestPosts(PAGEABLE)
 
-        then:
-        // not moderated posts first ...
+        then: 'not moderated posts first'
```
Comment (tdziurko) [inline src/test/groovy/com/jvm_bloggers/entities/blog_post/BlogPostRepositorySpec.groovy]: :clap:

File: src/test/groovy/com/jvm_bloggers/entities/blog_post/BlogPostRepositorySpec.groovy (hunk 4 above)
Comment (kepinskik) [reply src/test/groovy/com/jvm_bloggers/entities/blog_post/BlogPostRepositorySpec.groovy]: True, we don't. I thought the intention is more clear this way and consistent with the rest of them.

**Pull request merged**